pkg/operator/predicate/extension.go

```diff
@@ -29,7 +29,8 @@ func ExtensionRequirementsChanged() predicate.Predicate {
 			requiredRuntimeCondition := v1beta1helper.GetCondition(ext.Status.Conditions, operatorv1alpha1.ExtensionRequiredRuntime)
 			oldRequiredRuntimeCondition := v1beta1helper.GetCondition(oldExt.Status.Conditions, operatorv1alpha1.ExtensionRequiredRuntime)
 
-			if oldRequiredRuntimeCondition == nil && requiredRuntimeCondition != nil {
+			if oldRequiredRuntimeCondition == nil && requiredRuntimeCondition != nil ||
+				oldRequiredRuntimeCondition != nil && requiredRuntimeCondition == nil {
```

Comment:
While this probably works, do you mind adding brackets to make it more clear/readable?